marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/data-services/models/deleteModel.sjs

```diff
@@ -19,8 +19,23 @@ const ds = require("/data-hub/5/data-services/ds-utils.sjs");
 const entityLib = require("/data-hub/5/impl/entity-lib.sjs");
 
 var entityName;
-if (entityName === undefined || entityName === null) {
+if (!entityName) {
   ds.throwBadRequest("Must specify a name in order to delete an entity model");
 }
 
+const entityModel = entityLib.findModelByEntityName(entityName);
+if (!entityModel) {
+  ds.throwServerError(`Could not find entity model with name: ${entityName}`);
+}
+
+const entityTypeId = entityLib.getEntityTypeId(entityModel, entityName);
+const entityModelUri = entityLib.getModelUri(entityName);
+
+const steps = entityLib.findModelReferencesInSteps(entityName, entityTypeId);
+if (steps.length) {
+  ds.throwServerError(`Cannot delete the entity type '${entityName}' because it is referenced by the following step names: ${steps}`);
+}
+
 entityLib.deleteModel(entityName);
+entityLib.deleteModelReferencesInOtherModels(entityModelUri, entityTypeId);
+entityLib.deleteModelRelatedTDE(entityName);
```

Comment:
Moved TDE deletion from the trigger to this data service.

The previous TDE deletion code in the trigger module wasn't working as the URL for trigger document needs the model version which can only be obtained from the entity model that was originally deleted.
In order to make that work, we would have needed to perform a point in time query to fetch the entity model using something like this: `xdmp:request-timestamp()-1`.
As this is a post-commit trigger the task would be queued in the task server and run sometime after the original transaction commits. Since request timestamp value refers to when the task server executes the task and not the original deletion request, there wouldn’t be a reliable way to fetch the file in time.